dot15d4/src/csma/mod.rs

```diff
@@ -1,12 +1,909 @@
-use crate::phy::Device;
+pub mod constants;
+pub mod transmission;
+pub mod user_configurable_constants;
+mod utils;
 
-pub struct CsmaDevice<D: Device> {
-    device: D,
+use constants::*;
+use embedded_hal_async::delay::DelayNs;
+use rand_core::RngCore;
+use user_configurable_constants::*;
+
+use crate::{
+    frame::{Address, Frame, FrameBuilder, FrameRepr, FrameType},
+    phy::{
+        config::{RxConfig, TxConfig},
+        driver::{self, Driver, PacketBuffer},
+        radio::{
+            futures::{receive, transmit},
+            Radio, RadioFrame, RadioFrameMut, TxToken,
+        },
+    },
+    sync::{
+        channel::{Channel, Receiver, Sender},
+        join,
+        mutex::Mutex,
+        select,
+        yield_now::yield_now,
+        Either,
+    },
+    time::Duration,
+};
+
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Debug, Clone, Copy)]
+enum TransmissionTaskError<D: core::fmt::Debug> {
+    InvalidIEEEFrame,
+    InvalidDeviceFrame(D),
+}
+
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct CsmaConfig {
+    /// All to be transmitted frames will get the ack_request flag set if they are unicast and a data frame
+    ack_unicast: bool,
+    /// All to be transmitted frames will get the ack_request flag set if they are broadcast and a data frame
+    ack_broadcast: bool,
+    /// If false, all incoming packets will be sent up the layer stack, useful if making a sniffer. This does not include MAC layer control traffic
+    ignore_not_for_us: bool,
+    /// Even if there is no ack_request flag set, ack it anyway
+    ack_everything: bool,
+}
+
+impl Default for CsmaConfig {
+    fn default() -> Self {
+        Self {
+            ack_unicast: true,
+            ack_broadcast: false,
+            ignore_not_for_us: true,
```

Comment:
This is set `true`, however, the description of this field is _useful if making a sniffer_. Shouldn't this then be `false`?

Reply:
The idea here was that when `ignore_not_for_us==true`, all frames that are not broadcast or have a destination that is not corresponding to the current mote, would be ignored. Setting this to `false` would be more interesting for a sniffer, as this filter would be turned off.

I will add better documentation for this.